cancellation.go

```diff
@@ -0,0 +1,8 @@
+package ticketswitch
+
+type CancellationResult struct {
```

Comment:
how is error handling being done in cancellation call, just curious

Reply:
The error handling is complex. It's in the docs change that I've been working on. There are a few different cases, but essentially if you must cancel other items for the cancellation to work then you get the `MustAlsoCancel`. Otherwise you just have to check the updated `cancellation_status` at the order level.

Reply:
I was thinking about this and figured that I should probably just implement a function to check if the transaction is fully cancelled.